crates/ruff_linter/src/rules/pyupgrade/rules/native_literals.rs

```diff
@@ -215,20 +228,20 @@ pub(crate) fn native_literals(
                 return;
             }
 
+            if unary {
+                if !matches!(literal_type, LiteralType::Int | LiteralType::Float) {
+                    return;
+                }
```

Comment:
Would you mind adding a comment why this check is necessary for `unary` only and add a test case that verifies that ruff early returns if it's an unary and neither an `Int` nor `Float`

Reply:
I moved this up into the match.